test/src/specs/sync/ibd_process.rs

```diff
@@ -81,3 +81,101 @@ impl Spec for IBDProcess {
         assert!(is_node_sync, "node 5-6 must sync with node0");
     }
 }
+
+pub struct IBDProcessWithWhiteList;
+
+impl Spec for IBDProcessWithWhiteList {
+    crate::name!("ibd_process_with_whitelist");
+
+    crate::setup!(num_nodes: 7, connect_all: false);
+
+    fn run(&self, mut net: Net) {
+        info!("Running IBD process with whitelist");
+
+        {
+            let node6_listen = format!(
+                "/ip4/127.0.0.1/tcp/{}/p2p/{}",
+                net.nodes[6].p2p_port(),
+                net.nodes[6].node_id()
+            );
+
+            net.nodes[0].stop();
+
+            // whitelist will be connected on outbound on node start
+            net.nodes[0].start(
+                Box::new(|_| ()),
+                Box::new(move |config| {
+                    config.network.whitelist_peers = vec![node6_listen.parse().unwrap()]
+                }),
+            );
+        }
+
+        let node0 = &net.nodes[0];
+        let node1 = &net.nodes[1];
+        let node2 = &net.nodes[2];
+        let node3 = &net.nodes[3];
+        let node4 = &net.nodes[4];
+        let node5 = &net.nodes[5];
+        let node6 = &net.nodes[6];
+
+        node0.connect(node1);
+        node0.connect(node2);
+        node0.connect(node3);
+        node0.connect(node4);
+
+        // will never connect
```

Comment:
I am confused by this comment, why it will never connect?

Reply:
Because it is not protected/whitelist node and it is on ibd mode, so, the outbound session will be disconnected by node0

Reply:
I see, it's better to use the explicit constant and comments https://github.com/nervosnetwork/ckb/blob/da8897dbc8382293bdf8fadea380a0b79c1efa92/sync/src/lib.rs#L57 to offload the mind.